src/router/legacyPermalinkManagement.routerPlugin.js

```diff
@@ -158,9 +160,9 @@ const handleLegacyParam = (
         // When receiving a query, the application will encode the URI components
         // We decode those so that the new query won't encode encoded character
         // for example, we avoid having " " becoming %2520 in the URI
-        newQuery[key] = decodeURIComponent(newValue)
+        newQuery[key] = alreadyDecoded ? newValue : decodeURIComponent(newValue)
```

Comment:
,I would replace the 'alreadyDecoded' boolean by a `param==='layers'` check here, and integrate the line 114 explanation comment in the comment above. 
Unless there are other legacy parameters handling which decode the URI, of course, in which case we'll have to assign true to `alreadyDecoded` in those case instead.